src/NHibernate.Test/Ado/GenericBatchingBatcherFixture.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using NHibernate.AdoNet;
+using NHibernate.Cfg;
+using NHibernate.Dialect;
+using NUnit.Framework;
+using Environment = NHibernate.Cfg.Environment;
+
+namespace NHibernate.Test.Ado
+{
+	[TestFixture]
+	public class GenericBatchingBatcherFixture : TestCase
+	{
+		protected override string MappingsAssembly => "NHibernate.Test";
+
+		protected override IList Mappings => new[] {"Ado.VerySimple.hbm.xml"};
+
+		protected override void Configure(Configuration configuration)
+		{
+			configuration.SetProperty(Environment.BatchStrategy, typeof(GenericBatchingBatcherFactory).AssemblyQualifiedName);
+			configuration.SetProperty(Environment.GenerateStatistics, "true");
+			configuration.SetProperty(Environment.BatchSize, "1000");
+		}
+
+		protected override bool AppliesTo(Dialect.Dialect dialect)
+		{
+			return !(dialect is FirebirdDialect) &&
+			       !(dialect is Oracle8iDialect) &&
+			       !(dialect is MsSqlCeDialect);
+		}
+
+		[Test]
+		public void MassiveInsertUpdateDeleteTest()
+		{
+			var totalRecords = 1000;
+			BatchInsert(totalRecords);
+			BatchUpdate(totalRecords);
+			BatchDelete(totalRecords);
+
+			DbShoudBeEmpty();
+		}
+
+		[Test]
+		public void BatchSizeTest()
+		{
+			using (var sqlLog = new SqlLogSpy())
+			using (var s = Sfi.OpenSession())
+			using (var tx = s.BeginTransaction())
+			{
+				s.SetBatchSize(5);
+				for (var i = 0; i < 20; i++)
+				{
+					s.Save(new VerySimple { Id = 1 + i, Name = $"Fabio{i}", Weight = 1.45 + i });
+				}
+				tx.Commit();
+
+				var log = sqlLog.GetWholeLog();
+				Assert.That(4, Is.EqualTo(FindAllOccurrences(log, "Batch commands:")));
```

Comment:
Looks a bit inverted, I would instead have written this:
```c#
Assert.That(FindAllOccurrences(log, "Batch commands:"), Is.EqualTo(4));
```

Reply:
Fixed.